File: domain/src/main/java/co/joebirch/bastion/domain/interactor/GetTournaments.java
package co.joebirch.bastion.domain.interactor;

import java.util.List;

import javax.inject.Inject;

import co.joebirch.bastion.domain.executor.PostExecutionThread;
import co.joebirch.bastion.domain.executor.ThreadExecutor;
import co.joebirch.bastion.domain.model.Tournament;
import co.joebirch.bastion.domain.repository.TournamentsRepository;
import io.reactivex.Single;

public class GetTournaments extends UseCase<List<Tournament>, Void> {

    private final TournamentsRepository tournamentsRepository;

    @Inject
    GetTournaments(TournamentsRepository tournamentsRepository,
                   ThreadExecutor threadExecutor,
                   PostExecutionThread postExecutionThread) {
        super(threadExecutor, postExecutionThread);
        this.tournamentsRepository = tournamentsRepository;
    }

    @Override
    public Single<List<Tournament>> buildUseCaseObservable(Void unused) {
        return tournamentsRepository.getTournaments();
    }

}

File: domain/src/main/java/co/joebirch/bastion/domain/interactor/GetTournaments.kt
package co.joebirch.bastion.domain.interactor

import co.joebirch.bastion.domain.executor.PostExecutionThread
import co.joebirch.bastion.domain.executor.ThreadExecutor
import co.joebirch.bastion.domain.model.Tournament
import co.joebirch.bastion.domain.repository.TournamentsRepository
import io.reactivex.Single
import javax.inject.Inject

class GetTournaments @Inject constructor(private val tournamentsRepository: TournamentsRepository,
                     postExecutionThread: PostExecutionThread,
                     threadExecutor: ThreadExecutor)
    : UseCase<List<Tournament>, Void>(postExecutionThread, threadExecutor) {

    override fun buildUseCaseObservable(params: Void?): Single<List<Tournament>> {
        return tournamentsRepository.getTournaments()
    }

}